Framework/API/src/AlgorithmFactory.cpp

```diff
@@ -44,35 +44,47 @@ std::shared_ptr<Algorithm>
 AlgorithmFactoryImpl::create(const std::string &name,
                              const int &version) const {
   int local_version = version;
-  if (version < 0) {
-    if (version == -1) // get latest version since not supplied
-    {
-      auto it = m_vmap.find(name);
-      if (!name.empty()) {
-        if (it == m_vmap.end())
-          throw std::runtime_error("Algorithm not registered " + name);
-        else
-          local_version = it->second;
-      } else
-        throw std::runtime_error(
-            "Algorithm not registered (empty algorithm name)");
+  // Version not supplied
+  if (version == -1) {
+    auto v_it = m_vmap.find(name);
+    if (v_it == m_vmap.end()) {
+      // Try finding version from alias instead
+      auto a_it = m_amap.find(name);
```

Comment:
Could you help me understand how the `m_vmap` could have a version number from the alias name? As far as I can see the `m_vmap` is only ever filled with versions from the original class name?